src/odemis/driver/technolution.py

```diff
@@ -136,17 +168,174 @@ def ASM_API_Post_Call(self, url, expected_status, data=None, raw_response=False,
+        descanner.rotation.subscribe(self._startCalibration)
+        descan_offset = descanner.scanOffset.value
+        descanner.scanOffset.subscribe(self._startCalibration)
+        descanner.scanGain.subscribe(self._startCalibration)
+
+        # Retrieving and subscribing the scanner VA's
+        dwell_time = scanner.getTicksDwellTime
+        scanner.dwellTime.subscribe(self._startCalibration)
+        scan_rotation = scanner.rotation.value
+        scanner.rotation.subscribe(self._startCalibration)
+        scan_delay = scanner.scanDelay.value
+        scanner.scanDelay.subscribe(self._startCalibration)
+        scan_offset = scanner.scanOffset.value
+        scanner.scanOffset.subscribe(self._startCalibration)
+        scanner.scanGain.subscribe(self._startCalibration)
+
+        # Creation of setpoints
+        calibration_frequency = self.calibrationFrequency.value  # Frequency of the calibration signal in Hz
+        self.calibrationFrequency.subscribe(self._startCalibration)
+
+        time_points = numpy.arange(0, 100 / calibration_frequency, descanner.clockPeriod.value)
+        x_descan_setpoints = descanner.scanGain.value[0] * numpy.sin(2 * numpy.pi * calibration_frequency * time_points)
+        y_descan_setpoints = descanner.scanGain.value[1] * signal.sawtooth(2 * numpy.pi * calibration_frequency *
+                                                                           time_points)
+        x_scan_setpoints = scanner.scanGain.value[0] * numpy.sin(2 * numpy.pi * calibration_frequency * time_points)
+        y_scan_setpoints = scanner.scanGain.value[1] * signal.sawtooth(2 * numpy.pi * calibration_frequency *
+                                                                       time_points)
+
+        callibration_data = CalibrationLoopParameters(descan_rotation,
+                                                      descan_offset[0],
+                                                      x_descan_setpoints.astype(int).tolist(),
+                                                      descan_offset[1],
+                                                      y_descan_setpoints.astype(int).tolist(),
+                                                      dwell_time,
+                                                      scan_rotation,
+                                                      scan_delay[0],
+                                                      scan_offset[0],
+                                                      x_scan_setpoints.astype(int).tolist(),
+                                                      scan_offset[1],
+                                                      y_scan_setpoints.astype(int).tolist())
+
+        self.ASM_API_Post_Call("/scan/start_calibration_loop", 204, data=callibration_data.to_dict())
+
+    def _stopCalibration(self):
+        """
+        Unsubscribe _startCalibration from all the VA's so the calibration loop is not restarted when a VA is changed.
```

Comment:
That is actually what happens. The loop itself is not stopped in a sense that the calibration mode changes. But because the method startCalibration subscribes itself to all the VA's, these are reloaded again when one of them is changed and a new calibration loop is started with the updated parameters from the VA's.

Reply:
My comment on the .calibrationMode VA was incorrect. That is only used to decide if we perform a megafield acq or the calibration.

However, what I am worried about is the subscribe/unsubscribe implementation of the VAs to update values:
With the current implementation I think you risks that for example you call "startCalibration" again, while it is still executed. This could for example lead to issues with uploading the parameters to the ASM. If multiple VAs are changed, there would be multiple times that "startCalibration" is called about at the same time and each time you would do a upload request to the ASM. Even if there are no issues with accessing the same variable etc., you would still loose a lot of time by uploading the parameters to the ASM. 
We had something similar in the polarimetry. It was really hard to find the problem and in the end we solved it with a lock to make sure the method is never called simultaneously. We could to this, but this still would leave us with unnecessary calls to the ASM to upload parameters. But maybe that would be ok as for the calibration we do not care so much about time.

Reply:
Maybe we can just leave this be for now and if this if becoming a time issue we can fix it then.
BTW I added test cases where multiple VA's are changed multiple times (and also directly after each other) and that runs within 174 ms using the simulator.

Reply:
We want to make an appointment with Eric regarding this topic --> TODO Kornee